File: java/template-renderer/src/main/java/com/rbleggi/templaterenderer/renderers/HTMLRenderer.java
package com.rbleggi.templaterenderer.renderers;

import com.rbleggi.templaterenderer.core.TemplateRenderer;

import java.util.Map;

public class HTMLRenderer extends TemplateRenderer {
    @Override
    public byte[] render(Map<String, String> data) {
        System.out.println("Generating HTML Template...");

        var htmlContent = """
            <html>
            <head>
                <title>%s</title>
            </head>
            <body>
                <h1>%s</h1>
                <p>%s</p>
            </body>
            </html>
            """.formatted(data.get("title"), data.get("title"), data.get("content"));

        System.out.println("HTML Template Generated Successfully!");

        return htmlContent.getBytes();
    }
}

File: kotlin/template-renderer/src/main/kotlin/com/rbleggi/templaterenderer/renderers/HTMLRenderer.kt
package com.rbleggi.templaterenderer.renderers

import com.rbleggi.templaterenderer.core.TemplateRenderer

class HTMLRenderer : TemplateRenderer() {
    override fun render(data: Map<String, String>): ByteArray {
        println("Generating HTML Template...")

        val htmlContent = """
            <html>
            <head>
                <title>${data["title"]}</title>
            </head>
            <body>
                <h1>${data["title"]}</h1>
                <p>${data["content"]}</p>
            </body>
            </html>
        """.trimIndent()

        println("HTML Template Generated Successfully!")

        return htmlContent.toByteArray()
    }
}